log.js

```diff
@@ -248,3 +217,31 @@ log.addLevel('silent', Infinity)
 
 // allow 'error' prefix
 log.on('error', function(){})
+
+// We want the log object to share the methods from the tracker,
+// and vice-versa, so mix them into each other.
+var trackerConstructors = ['newGroup', 'newItem', 'newStream']
+
+// mixin the public methods from log into the tracker
+// (except: conflicts and one's we handle specially)
+Object.keys(log).forEach(function (P) {
+  if (P[0] === '_') return
+  if (trackerConstructors.filter(function (C) { return C === P }).length) return
+  if (typeof log[P] !== 'function') return
```

Comment:
Yep, it would, that's the original code. I'm working on a way to amend this
PR (and have already changed it there) without clobbering the module for
other consumers but its a decent amount of work.
On Jan 28, 2016 10:47 PM, "David Fox-Powell" <notifications@github.com>
wrote:

> In log.js <https://github.com/npm/npmlog/pull/26#discussion_r51225600>:
>
> > @@ -248,3 +217,31 @@ log.addLevel('silent', Infinity)
> >
> >  // allow 'error' prefix
> >  log.on('error', function(){})
> > +
> > +// We want the log object to share the methods from the tracker,
> > +// and vice-versa, so mix them into each other.
> > +var trackerConstructors = ['newGroup', 'newItem', 'newStream']
> > +
> > +// mixin the public methods from log into the tracker
> > +// (except: conflicts and one's we handle specially)
> > +Object.keys(log).forEach(function (P) {
> > +  if (P[0] === '_') return
> > +  if (trackerConstructors.filter(function (C) { return C === P }).length) return
> > +  if (typeof log[P] !== 'function') return
>
> Why filter inside forEach. Shouldn't an indexOf check work here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/npm/npmlog/pull/26/files#r51225600>.
>